src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -172,16 +172,32 @@ internal string GetDebugString()
             return sb.ToString();
         }
 
+        private void AppendStorageProviderValuesToDebugString(StringBuilder sb)
+        {
+            sb.Append(nameof(this.Storage)).Append(": { ");
+            foreach (var value in this.Storage)
+            {
+                sb.Append(value.Key).Append(": ").Append(value.Value).Append(", ");
+            }
+
+            sb.Append(" }, ");
+        }
+
         /// <summary>
         /// A helper method to help retrieve the connection string name for the configured storage provider.
         /// </summary>
         /// <returns>The connection string name for the configured storage provider.</returns>
         public string GetConnectionStringName()
         {
-            return this.StorageProvider.GetConfiguredProvider().ConnectionStringName;
+            if (this.Storage.ContainsKey("connectionStringName"))
```

Comment:
I assume this is specific to the Azure Storage provider. How will the Event Hubs provider specify its connection string? I have many more questions about this, but I think there are some fundamental design pieces that I need to understand first.

Reply:
I thought about this a bit more and believe I have addressed this in my latest iteration. I will add clarification in the morning of how this works.

Reply:
This is now more clearly handled by the factory/storage provider. The factory provides the connection name to the storage provider, and then the default storage provider instance's connection name is used to build our HTTP API payload, which is the only place this method was used.